lib/src/app.dart

```diff
@@ -1,17 +1,39 @@
 import 'satellites.dart';
 import 'package:flutter/material.dart';
 import '../screens/home.dart';
+import 'satellite.dart';
 
 class SatTrack extends StatelessWidget {
-  Satellites satData;
-  SatTrack(this.satData, {Key? key}) : super(key: key);
+  SatTrack({Key? key}) : super(key: key);
 
   @override
   Widget build(BuildContext context) {
-    return MaterialApp(
-      title: 'Sat Track',
-      theme: ThemeData(primarySwatch: Colors.blueGrey),
-      home: HomeScreen(satData),
+    return FutureBuilder<Satellites>(
+      future: satData,
+      builder: (context, AsyncSnapshot<Satellites> snapshot) {
+        if (snapshot.hasData) {
+          return MaterialApp(
+            title: 'Sat Track',
+            theme: ThemeData(primarySwatch: Colors.blueGrey),
+            home: HomeScreen(snapshot.data!),
+          );
+        } else {
+          return CircularProgressIndicator();
```

Comment:
I have not built this yet.  But when you load this, I assume because the satData() is async, that the widgets are built before the data is returned.  Meaning progress indicator is shown at load...  Once the data is loaded into your satsAbove() object you will have to refresh the app state() to cause a reload right?  Does this just stick on "CircularProgress()" forever?

Reply:
It doesn't do it forever. The FutureBuild widget is really great -- it runs the "else" branch until the future part is complete (i.e. the "snapshot" has data). So it'll keep checking for the snapshot to finish, then loads the homepage.

Reply:
Changes everything I thought I knew about state. lol. That means the future is marking that piece as dirty inside of a stateless widget.  I assume it only rebuilds the portion in the future wrapper. Cool stuff